File: greenbank-spigot/src/main/java/org/flameyosflow/greenbank/GreenBankSupport.java
package org.flameyosflow.greenbank;

import org.bukkit.event.Listener;
import org.bukkit.plugin.java.JavaPlugin;

import org.flameyosflow.greenbank.commands.GreenBankCommand;

import org.jetbrains.annotations.NotNull;

public class GreenBankSupport extends JavaPlugin {

    /*
     * getVersion() returns the version of the plugin.
     *
     * (x is a number)
     * x.x.x is the initial version of the plugin
     * build x says how many times the provided initial version has been tested
     *
     * for example: 1.0.0 build 6 (1.0.0 was tested 6 times)
     * another example: 1.0.1 build 4 (1.0.1 was tested 4 times)
     *
     * every little different version has different number of builds.
     */
    public String getVersion() {
        return "1.0.0 BUILD 10";
    }

    protected void registerCommand(String name, GreenBankCommand command) {
        GreenBankCommand.register(name, command);
    }

    protected void registerListener(@NotNull Listener listener, @NotNull JavaPlugin plugin) {
        getServer().getPluginManager().registerEvents(listener, plugin);
    }
}

File: greenbank-spigot/src/main/java/org/flameyosflow/greenbank/GreenBankSupport.kt
package org.flameyosflow.greenbank

import org.bukkit.Bukkit
import org.bukkit.event.Listener
import org.bukkit.plugin.java.JavaPlugin

import org.flameyosflow.greenbank.commands.*
import org.flameyosflow.greenbank.listeners.*
import org.flameyosflow.greenbank.commands.noteconomy.GreenbankCommand

open class GreenBankSupport : JavaPlugin() {
    // get version from build.gradle.kts
    val version: String = "1.0.0 build 21"

    private fun registerCommand(name: String, command: GreenBankCommand?, greenBank: GreenBankMain) {
        val command2 = Bukkit.getPluginCommand(name) ?: return
        command2.setExecutor(command)

        // add command to internal register
        if (greenBank.configFile!!.getBoolean("override-all-other-plugins")) CommandSupport.add(name, command2)
    }
    private fun registerListener(listener: Listener, plugin: JavaPlugin) = server.pluginManager.registerEvents(listener, plugin)

    protected fun registerAll(greenBank: GreenBankMain) {
        registerCommand("balance", BalanceCommand(greenBank), greenBank); registerCommand("pay", PayCommand(greenBank), greenBank)
        registerCommand("eco", EcoCommand(greenBank), greenBank); registerCommand("greenbank", GreenbankCommand(greenBank), greenBank)
        registerListener(JoinListener(greenBank), greenBank as JavaPlugin)
    }
}